insights/client/constants.py

```diff
@@ -1,8 +1,42 @@
 import os
 
+_user_home = os.path.expanduser('~')
+_app_name = 'insights-client'
+_uid = os.getuid()
+
+
+def _log_dir():
+    '''
+    Get the insights-client log dir
+
+    Default: /var/log/insights-client
+    Non-root user: $XDG_CACHE_HOME/insights-client || $HOME/.cache/insights-client
+    '''
+    if _uid == 0:
+        insights_log_dir = os.path.join(os.sep, 'var', 'log', _app_name)
+    else:
+        local_log_dir = os.getenv('XDG_CACHE_HOME', default=os.path.join(_user_home, '.cache'))
```

Comment:
Nice use of the XDG base directory specification. I looked at what options there are in the specification. Looks like we have 3 to choose from:

* `XDG_CONFIG_HOME`: user-specific configuration files
* `XDG_CACHE_HOME`: user-specific non-essential (cached) data
* `XDG_DATA_HOME`: user-specific data files

There isn't really a set of XDG directories that directly map to the various high-level directories in the FHS (`/var`, `/run`, `/usr/share`, `/etc`, etc.). I think it makes sense to put the logs under `~/.cache`. Does it make sense to also put the downloaded eggs into `~/.cache`? What would happen to a registered client if the latest.egg suddenly disappeared? Would it simply download a new egg again? If so, I think it could be reasoned to be "non-essential data". This way we aren't deviating from the specification of XDG base directories and creating our own hierarchy.